diff --git a/src/backend/utils/adt/date.c b/src/backend/utils/adt/date.c
index 4227ab1a72b..c9ea891883f 100644
--- a/src/backend/utils/adt/date.c
+++ b/src/backend/utils/adt/date.c
@@ -3128,7 +3128,7 @@ timetz_zone(PG_FUNCTION_ARGS)
 	TimeTzADT  *t = PG_GETARG_TIMETZADT_P(1);
 	TimeTzADT  *result;
 	int			tz;
-	char		tzname[TZ_STRLEN_MAX + 1];
+	char		tzname_[TZ_STRLEN_MAX + 1];
 	int			type,
 				val;
 	pg_tz	   *tzp;
@@ -3138,7 +3138,7 @@ timetz_zone(PG_FUNCTION_ARGS)
 	 */
 	text_to_cstring_buffer(zone, tzname, sizeof(tzname));
 
-	type = DecodeTimezoneName(tzname, &val, &tzp);
+	type = DecodeTimezoneName(tzname_, &val, &tzp);
 
 	if (type == TZNAME_FIXED_OFFSET)
 	{
