diff --git a/src/Makefile.global.in b/src/Makefile.global.in
index 04952b533de..15a77f6c0ad 100644
--- a/src/Makefile.global.in
+++ b/src/Makefile.global.in
@@ -789,7 +789,8 @@ LIBOBJS = @LIBOBJS@
 # files needed for the chosen CRC-32C implementation
 PG_CRC32C_OBJS = @PG_CRC32C_OBJS@
 
-LIBS := -lpgcommon -lpgport $(LIBS)
+#LIBS := -lpgcommon -lpgport $(LIBS)
+LIBS := -L$(top_builddir)/src/common -lpgcommon -L$(top_builddir)/src/common -lpgcommon_shlib -L$(top_builddir)/src/common -lpgcommon_srv -L$(top_builddir)/src/port -lpgport -L$(top_builddir)/src/port -lpgport_shlib -L$(top_builddir)/src/port -lpgport_srv -lzoslib -lzoslib-supp -lz
 
 # to make ws2_32.lib the last library
 ifeq ($(PORTNAME),win32)
