diff --git a/src/Makefile.shlib b/src/Makefile.shlib
index fa81f6ffdd6..ae8d01edd6c 100644
--- a/src/Makefile.shlib
+++ b/src/Makefile.shlib
@@ -183,6 +183,19 @@ ifeq ($(PORTNAME), linux)
   endif
 endif
 
+ifeq ($(PORTNAME), zos)
+  LINK.shared   	= $(COMPILER)
+  ifdef SHLIB_EXPORTS
+    exports_file	= $(SHLIB_EXPORTS:%.txt=%.x)
+#    $(exports_file): $(SHLIB_EXPORTS)
+#    echo 'EXPORTS' > $@
+#    $(AWK) '/^[^\#]/ {print $$1}' $< >> $@
+#    LINK.shared	 += -Wl,export,$(exports_file)
+  else
+#    LINK.shared         += -Wl,export_all
+  endif
+endif
+
 ifeq ($(PORTNAME), solaris)
   LINK.shared		= $(COMPILER) -shared
   ifdef soname
@@ -256,9 +269,20 @@ endif #haslibarule
 
 ifeq (,$(filter cygwin win32,$(PORTNAME)))
 
+ifeq ($(PORTNAME), zos)
+OBJFILES_TXT=$(shell find $(top_builddir)/src/ -name objfiles.txt)
+OBJS_NEW=$(strip \
+  $(foreach file,$(OBJFILES_TXT), \
+    $(shell tr ' \t\r\n' '\n\n\n\n' < $(file) | awk 'NF') \
+  ) \
+)
+OBJS_NEW_1=$(foreach obj,$(OBJS_NEW),$(top_builddir)/$(obj))
+OBJS+=$(OBJS_NEW_1)
+endif
+
 # Normal case
 $(shlib): $(OBJS) | $(SHLIB_PREREQS)
-	$(LINK.shared) -o $@ $(OBJS) $(LDFLAGS) $(LDFLAGS_SL) $(SHLIB_LINK)
+	$(LINK.shared) -o $@ $(OBJS) $(LIBS) $(LDFLAGS) $(LDFLAGS_SL) $(SHLIB_LINK) $(SHLIB_LINK_INTERNAL) -Wl,-bMSGLEVEL=8
 ifdef shlib_major
 # If we're using major and minor versions, then make a symlink to major-version-only.
 ifneq ($(shlib), $(shlib_major))
