diff --git a/src/include/pgtime.h b/src/include/pgtime.h
index 5fc9f223de3..a782776f41b 100644
--- a/src/include/pgtime.h
+++ b/src/include/pgtime.h
@@ -91,7 +91,7 @@ extern PGDLLIMPORT pg_tz *session_timezone;
 extern PGDLLIMPORT pg_tz *log_timezone;
 
 extern void pg_timezone_initialize(void);
-extern pg_tz *pg_tzset(const char *tzname);
+extern pg_tz *pg_tzset(const char *tzname_);
 extern pg_tz *pg_tzset_offset(long gmtoffset);
 
 extern pg_tzenum *pg_tzenumerate_start(void);
