diff --git a/src/timezone/pgtz.c b/src/timezone/pgtz.c
index 671b4d76237..d74b1d2aa42 100644
--- a/src/timezone/pgtz.c
+++ b/src/timezone/pgtz.c
@@ -321,7 +321,7 @@ pg_tzset_offset(long gmtoffset)
 {
 	long		absoffset = (gmtoffset < 0) ? -gmtoffset : gmtoffset;
 	char		offsetstr[64];
-	char		tzname[128];
+	char		tzname_[128];
 
 	snprintf(offsetstr, sizeof(offsetstr),
 			 "%02ld", absoffset / SECS_PER_HOUR);
@@ -338,13 +338,13 @@ pg_tzset_offset(long gmtoffset)
 					 ":%02ld", absoffset);
 	}
 	if (gmtoffset > 0)
-		snprintf(tzname, sizeof(tzname), "<-%s>+%s",
+		snprintf(tzname_, sizeof(tzname_), "<-%s>+%s",
 				 offsetstr, offsetstr);
 	else
-		snprintf(tzname, sizeof(tzname), "<+%s>-%s",
+		snprintf(tzname_, sizeof(tzname_), "<+%s>-%s",
 				 offsetstr, offsetstr);
 
-	return pg_tzset(tzname);
+	return pg_tzset(tzname_);
 }
 
 
